segment.go

```diff
@@ -237,7 +237,11 @@ func newSegmentRoot(segmentSpan *segmentSpanImpl) *rootSegmentSpan {
 				break
 			}
 		}
-		s.tracer.reporter.Send(append(s.segment, s))
+		//check need send segment with sample service
+		sampled := s.tracer.sampler.IsSampled(s.TraceID, s.FirstSpan.GetOperationName())
```

Comment:
1. Should be sampled when the first `Span` is created, and `NoopSpan` is returned when ignored.
1. Sampling should be forced when downstream services are sampled.

Hi @xbkaishui, that's my suggestion, what do you think?

Reply:
Hi @arugal, thanks for your cr, I have a couple of questions about what your mention
for the first one, how to decide which one is the first span ? 
for the second one, how to check when downstream services are sampled, if it is have sw8, I should consider it is force sampled?

Reply:
> how to decide which one is the first span ?

https://github.com/SkyAPM/go2sky/blob/fdae247dbb32f87c064aa59e557d814c2eb62fb5/trace.go#L116-L119

When `parentSpan` is `nil`.

> how to check when downstream services are sampled, if it is have sw8, I should consider it is force sampled?

Yes, when `sw8` header exist.

 
I think that the code should be added to `trace#CreateLocalSpan` or `trace#createNoop`.

Reply:
Hi @arugal  Thanks for you help,  changed as your suggestion. please help review again

Reply:
1. `CreateEntrySpan` and `CreateExitSpan` should judge the `CreateLocalSpan` returns `NoopSpan` or not. Specifically `CreateExitSpan`, if `NoopSpan` returns `span type is wrong` error.

2. `traceID` is not required for `IsSampled`, please recheck.